v-next/hardhat-utils/src/crypto.ts

```diff
@@ -17,17 +15,22 @@ export async function keccak256(bytes: Uint8Array): Promise<Uint8Array> {
  *
  * This function is primarily intended for generating unique identifiers from
  * a given input string.
- * It uses the MD5 hash algorithm, which is not cryptographically secure, but
+ * It uses the SHA-1 hash algorithm, which is not cryptographically secure, but
  * is sufficient for this use case as long as the input is not generated by an
  * attacker.
  *
- * Note: The exact algorithm used (MD5) is not crucial for the function's
+ * Note: The exact algorithm used (SHA-1) is not crucial for the function's
  * purpose of generating unique identifiers, and could be replaced if needed.
  *
  * @param data The input string to be hashed.
- * @returns The MD5 hash of the input string, represented as a
+ * @returns The SHA-1 hash of the input string, represented as a
  * hexadecimal string.
  */
-export function createNonCryptographicHashId(data: string): string {
-  return createHash("md5").update(data).digest("hex");
+export async function createNonCryptographicHashId(
+  data: string,
+): Promise<string> {
+  const message = new TextEncoder().encode(data);
+  const buffer = await crypto.subtle.digest("SHA-1", message);
+  const array = Array.from(new Uint8Array(buffer));
+  return array.map((b) => b.toString(16).padStart(2, "0")).join("");
```

Comment:
```suggestion
  const buffer = await crypto.subtle.digest("SHA-1", message);
  return Buffer.from(buffer).toString("hex");
```

This is probably much faster, as it doesn't iterate in js.

Reply:
Oh nice! It totally makes sense. I haven't thought of that initially.